src/targets/gpu/allocation_model.cpp

```diff
@@ -30,18 +30,18 @@ namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace gpu {
 
-std::string gpu_allocation_model::name() const { return "hip::allocate"; }
-operation gpu_allocation_model::allocate(const shape& s) const
+std::string gpu_allocation_model::name() { return "hip::allocate"; }
+operation gpu_allocation_model::allocate(const shape& s)
 {
     return make_op(name(), {{"shape", to_value(s)}});
 }
 
-operation gpu_allocation_model::preallocate(const shape& s, const std::string& id) const
+operation gpu_allocation_model::preallocate(const shape& s, const std::string& id)
 {
     return make_op("hip::hip_allocate_memory", {{"shape", to_value(s)}, {"id", id}});
 }
 
-std::string gpu_allocation_model::copy() const { return "hip::copy"; }
+std::string gpu_allocation_model::copy() { return "hip::copy"; }
```

Comment:
Why is `const` being removed?

Reply:
Tool for static analysis suggested turning it to `static` or `constexpr` for better optimization.